tests/test_views.py

```diff
@@ -397,6 +398,31 @@ def test_ajax_upload_clipboardadmin_same_name_as_existing_file_in_moderation(sel
         error_msg = 'Cannot archive existing test1.jpg file version'
         self.assertEqual(response.json()['error'], error_msg)
 
+    def test_ajax_upload_clipboardadmin_xss_vulnerability(self):
+        """
+        If we add malicious data to an ajax upload request ensure it is stripped in response.
+        """
+        with self.login_user_context(self.superuser):
+            response = self.client.post(
+                reverse('admin:filer-ajax_upload'),
+                data={'path': '<script>alert("attack!")</script>'}
+            )
+
+        self.assertFalse('<script>alert("attack!")</script>' in response.request.values())
```

Comment:
What's actually in response.request.values()? Does it have the values url encoded?

Reply:
Request here is a dictionary representation of what was in the request. It strips that value and passes on the valid path value to the file being used.